topic.go

```diff
@@ -164,13 +164,15 @@ func (t *Topic) Publish(ctx context.Context, data []byte, opts ...PubOpt) error
 		return ErrTopicClosed
 	}
 
-	seqno := t.p.nextSeqno()
-	id := t.p.host.ID()
 	m := &pb.Message{
 		Data:     data,
 		TopicIDs: []string{t.topic},
-		From:     []byte(id),
-		Seqno:    seqno,
+		From:     nil,
+		Seqno:    nil,
+	}
+	if t.p.signID != "" {
+		m.From = []byte(t.p.signID)
+		m.Seqno = t.p.nextSeqno()
```

Comment:
Note: I don't think whether we set or not the `Seqno` is correlated with whether we send or not the source peer ID, rather with whether we have a custom `MessageIdFn` or not.

Reply:
Yes, we can introduce another option for it, but the thought here is that seq-no makes little sense without a message author (since everyone can claim the seq-no for any message data), so it's just left out.

There's some polishing that can be done. I appreciate the quick PR merge, but would have welcomed more feedback/discussion. Reviewing the update path for eth2 clients would also help. Implementations all behave a little different, and this is the chance to plan signing functionality (non-eth2, but relevant) for those that don't have it yet, make verification strict yes/no, and align everything. I am tracking behavioral differences in a table in the eth2-specs issue here: https://github.com/ethereum/eth2.0-specs/issues/1981#issuecomment-663079721

Reply:
Sure, that's why I haven't cut a release yet -- we can iterate on it.